### ISSUE TITLE
False positive for "except clause unreachable" when excepting variable annotated as `Union[Type[Exception], tuple[Type[Exception], ...]]`

### ISSUE THREAD

**Issue by timbedard** [microsoft/pyright#3270]
**Describe the bug**
When the annotation for a variable is `Union[Type[Exception], tuple[Type[Exception], ...]]`, `except`ing that variable after another `except` clause will cause Pyright to report that the variable except clause is unreachable.

**To Reproduce**
Annotate a variable as `Union[Type[Exception], tuple[Type[Exception], ...]]` and attempt to `except` it following another `except` clause.

Note: This does not occur when the type is annotated as `Type[Exception]` or `tuple[Type[Exception], ...]` on their own.

**Expected behavior**
The clause is not marked as unreachable.

**Screenshots or Code**
```python
def example(u: Union[Type[Exception], tuple[Type[Exception], ...]]):
    try:
        ...
    except ValueError as e:
        ...
    except u as e:  # "Except clause is unreachable because exception is already handled"
        ...
```

**VS Code extension or command-line**
command-line via Neovim LSP client

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.234, which I just published. It will also be included in the next pylance release.
